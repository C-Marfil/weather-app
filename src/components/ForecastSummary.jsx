import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";

const ForecastSummary = (props) => {
  const { date, temperature, description, icon, onSelect } = props;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__temperature">
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <button type="button" onClick={() => onSelect(date, description)}>
        More details
      </button>
    </div>
  );
};

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default ForecastSummary;
