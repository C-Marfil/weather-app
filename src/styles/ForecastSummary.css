.forecast-summary {
  background-color: rgb(70, 56, 56);
  color: #fff;
  opacity: 0.5;
  border-radius: 10px;
  padding: 10px;
  display: flex;
  flex-direction: column;
}
