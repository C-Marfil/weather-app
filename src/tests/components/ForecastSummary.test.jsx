import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1525046400000,
    description: "Stub description",
    icon: "800",
    temperature: {
      min: 4,
      max: 11,
    },
    onSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("it renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );

    expect(getByText("Mon Apr 30 2018")).toHaveAttribute(
      "class",
      "forecast-summary__date"
    );
    expect(getByText("Stub description")).toHaveAttribute(
      "class",
      "forecast-summary__description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("11Â°C")).toHaveAttribute(
      "class",
      "forecast-summary__temperature"
    );
  });
});
