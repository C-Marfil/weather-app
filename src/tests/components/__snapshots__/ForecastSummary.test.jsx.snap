// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Mon Apr 30 2018
    </div>
    <div
      class="forecast-summary__temperature"
    >
      11Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
  </div>
</DocumentFragment>
`;
